compiler/src/dotty/tools/dotc/ast/Trees.scala

```diff
@@ -725,6 +725,8 @@ object Trees {
     assert(tpt != genericEmptyTree)
     def unforced: LazyTree = preRhs
     protected def force(x: AnyRef): Unit = preRhs = x
+
+    def withName(name: TermName)(implicit src: SourceFile): DefDef[T] = copy(name = name)
```

Comment:
I think `copy` should never be called on trees as it will not copy over positions, types... All copy should go through a tree copier. @smarter correct me if I'm wrong.

If that's the case, we should probably find a way to enforce it somehow. I see that `copy` is currently used in a few places in the code base.

Reply:
Yes. The only reason I call it here is because it was previously called from outside of the `ast` package.

Reply:
Yup it's good thing copy isn't accessible with your changes :). I've left suggestions on how to rewrite the existing usages using a tree copier instead.

Reply:
I believe after your latest changes, this method can be removed